src/qt/test/optiontests.cpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <qt/bitcoin.h>
+#include <qt/test/optiontests.h>
+#include <test/util/setup_common.h>
+#include <util/system.h>
+
+#include <QSettings>
+#include <QTest>
+
+#include <univalue.h>
+
+//! Entry point for BitcoinApplication tests.
+void OptionTests::optionTests()
+{
+    // Test regression https://github.com/bitcoin/bitcoin/issues/24457. Ensure
+    // that setting integer prune value doesn't cause an exception to be thrown
+    // in the OptionsModel constructor
+    gArgs.LockSettings([&](util::Settings& settings) {
+        settings.forced_settings.erase("prune");
+        settings.rw_settings["prune"] = 3814;
+    });
+    gArgs.WriteSettingsFile();
+    OptionsModel{};
```

Comment:
84b0973e if you retouch, maybe separate the two sections in this function with a line break to clarify that the comment refers to the first section (prune=3814).